python/lsst/pipe/base/connections.py

```diff
@@ -456,46 +457,124 @@ def buildDatasetRefs(self, quantum: Quantum) -> typing.Tuple[InputQuantizedConne
                                      "in input quantum")
         return inputDatasetRefs, outputDatasetRefs
 
-    def adjustQuantum(self, datasetRefMap: NamedKeyDict[DatasetType, typing.Set[DatasetRef]]
-                      ) -> NamedKeyDict[DatasetType, typing.Set[DatasetRef]]:
+    def adjustQuantum(
+        self,
+        inputs: Iterable[typing.Tuple[str, BaseInput, typing.FrozenSet[DatasetRef]]],
+        label: str,
+        dataId: DataCoordinate,
+    ) -> typing.Iterator[typing.Tuple[str, BaseInput, typing.AbstractSet[DatasetRef]]]:
         """Override to make adjustments to `lsst.daf.butler.DatasetRef` objects
         in the `lsst.daf.butler.core.Quantum` during the graph generation stage
         of the activator.
 
-        The base class implementation simply checks that input connections with
-        ``multiple`` set to `False` have no more than one dataset.
-
         Parameters
         ----------
-        datasetRefMap : `NamedKeyDict`
-            Mapping from dataset type to a `set` of
-            `lsst.daf.butler.DatasetRef` objects
+        inputs : `Iterable` of `tuple`
+            Three-element tuples, each a connection name, the connection
```

Comment:
I was idly wondering this last night (this code was written a month or so ago and I hadn't looked at it for a long time), and I think it's a mistake: I was thinking that there was no viable unique key, because I was thinking that the connection name was a `DatasetType` name (and a task can have multiple connections resolve to the same dataset type, e.g. the same refcat for both astrometry and photometry).  But it is the internal connection name, and that should be unique within a task, and as I found myself transforming this into a mapping (from name to two-tuple of connections instance and list of refs) in one of the `pipe_tasks` `adjustQuantum` implementations, that does seem like a better choice.  I'll make that change and see how it goes.